git/file.go

```diff
@@ -72,6 +72,45 @@ func (f File) IsDir() bool {
 
 }
 
+// Returns true if the file is inside a submodule and the submodule file.
+func (f File) IsInSubmodule(c *Client) (bool, File, error) {
```

Comment:
Can you add a FIXME saying that when we implement submodules this should be inverted? (Submodules contain Files, not the other way around,  but we don't currently have a submodule type..)

Reply:
+1